src/Features/Core/Portable/GenerateMember/GenerateConstructor/AbstractGenerateConstructorService.Editor.cs

```diff
@@ -23,6 +23,38 @@ internal abstract partial class AbstractGenerateConstructorService<TService, TAr
 
         internal abstract IMethodSymbol GetDelegatingConstructor(State state, SemanticDocument document, int argumentCount, INamedTypeSymbol namedType, ISet<IMethodSymbol> candidates, CancellationToken cancellationToken);
 
+        protected abstract IMethodSymbol GetCurrentConstructor(SemanticModel semanticModel, SyntaxToken token, CancellationToken cancellationToken);
+
+        protected abstract IMethodSymbol GetDelegatedConstructor(SemanticModel semanticModel, IMethodSymbol constructor, CancellationToken cancellationToken);
+
+        protected bool CanDelegeteThisConstructor(State state, SemanticDocument document, IMethodSymbol delegatedConstructor, CancellationToken cancellationToken = default)
+        {
+            var constructorsCount = delegatedConstructor.ContainingType.Constructors.Length;
+            var currentConstructor = GetCurrentConstructor(document.SemanticModel, state.Token, cancellationToken);
+            if (currentConstructor == delegatedConstructor)
+            {
+                return false;
+            }
+
+            // We need ensure that delegating constructor won't cause circular dependency.
+            // The chain of dependency can not exceed the number for constructors
+            for (var i = 0; i < constructorsCount; i++)
+            {
+                delegatedConstructor = GetDelegatedConstructor(document.SemanticModel, delegatedConstructor, cancellationToken);
+                if (delegatedConstructor == null)
+                {
+                    return true;
+                }
+
+                if (delegatedConstructor == currentConstructor)
```

Comment:
❓ Shouldn't we be using `.Equals` to compare symbols?

/cc @CyrusNajmabadi who may know

Reply:
Yes.  we should.